nntrainer/models/neuralnet.cpp

```diff
@@ -205,11 +205,7 @@ int NeuralNetwork::initialize() {
 
   initialized = true;
 
-  // @note we need check loadedWeight for the case of multiple call of load to
-  // load weight. Only the weight needs to be loaded here. Becuase the buffer
-  // for the optimizer is not allocated yet.
-  // loadedWeight check is just for the duplicate load of weight.
-  if (!load_path.empty() && !loadedWeight) {
+  if (!load_path.empty()) {
```

Comment:
At this time, the opt variables are not allocated, I guess. correct me if I'm wrong.

Reply:
Yes it used to be, in this patch optimizer variable is allocated when weight is being allocated(manager.cpp:608), we may argue this is not the optimized strategy though in case of inference mode.